flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/JobTerminationMessageParameters.java

```diff
@@ -28,8 +28,8 @@
  */
 public class JobTerminationMessageParameters extends MessageParameters {
 
-	private final JobIDPathParameter jobPathParameter = new JobIDPathParameter();
-	private final TerminationModeQueryParameter terminationModeQueryParameter = new TerminationModeQueryParameter();
+	public final JobIDPathParameter jobPathParameter = new JobIDPathParameter();
+	public final TerminationModeQueryParameter terminationModeQueryParameter = new TerminationModeQueryParameter();
```

Comment:
so that you can actually access (and resolve) them in `RestClusterClient#stop()`.

Reply:
Hmm what about giving these values to the `JobTerminationMessageParameters` constructor? It feels not entirely right that we access internal fields to resolve them?

Reply:
well they aren't really internal fields, at least i didn't intend them to be. The client has to resolve the parameters somehow, so we either have to add a custom resolve method to every `MessageParameters` class (which will make for an odd API when creating sub-classes), or provide access to each parameter (either directly or through a getter). I opted for the direct approach since it makes it obvious that we are in fact modifying the `MessageParameters` object.

Reply:
we could also add a dedicated resolve method for each parameter that a particular class defines, but at that point we're duplicating the individual parameter methods.